### ISSUE TITLE
CSS3 module "Generated Content for Paged Media" and especially the target-counter function

### ISSUE THREAD

**Issue by Fyrd** [Fyrd/caniuse#1510]
CSS3 module &quot;Generated Content for Paged Media&quot; and especially the target-counter function

---------
##### This issue was imported from [Google Moderator](https://www.google.com/moderator/)
Moderator votes: __+3__
Issue added by __tynstar__ on __2014-08-07__
To vote this issue up or down, simply include `+1` or `-1` in your comment.

**Comment by DavidSabine** [comment]
+1

**Comment by puco** [comment]
+1

**Comment by ntlk** [comment]
Adding this would be extremely helpful.

**Comment by jrabbe** [comment]
Agreed, right now finding out what the support is for Paged Media in browsers is a pain.

**Comment by DiMFail** [comment]
+1

**Comment by jonathansnell** [comment]
+1

**Comment by tgr** [comment]
Current status: https://www.chromestatus.com/feature/5400251359821824 (not implemented anywhere)

**Comment by jaggzh** [comment]
PrinceXML https://www.princexml.com/ supports it.  (Note: I am not affiliated with them in any way, just using their software due to the HTML/CSS print features they support that nothing else seems to).

**Comment by tgr** [comment]
So does [Vivliostyle](https://github.com/vivliostyle) and probably a number of other [print tools](https://print-css.rocks/tools.html). But I don't believe caniuse tracks those.

**Comment by tgraham-antenna** [comment]
+1

Antenna House Formatter (https://www.antennahouse.com/antenna1/formatter/) supports GCPM as well.  I would, of course, also like to see caniuse track paged media formatters alongside browsers.

**Comment by JimSaiya** [comment]
+1

**Comment by Porges** [comment]
>  I would, of course, also like to see caniuse track paged media formatters alongside browsers.

Another category that should be added site-wide is EPUB/MOBI renderers (since it's all just HTML & CSS in the end). Often Amazon's documentation about what the Kindle supports is inaccurate.

**Comment by pl86** [comment]
+1

**Comment by stephanschubert** [comment]
[CONTRIBUTING.md](https://github.com/Fyrd/caniuse/blob/master/CONTRIBUTING.md) says it's currently not possible to add another browser.. Why's that?

I would also really like to add [KF8/KFX support](http://kindlegen.s3.amazonaws.com/AmazonKindlePublishingGuidelines.pdf) (see appendix 16.1+) as a common denominator for EPUB & MOBI.

**Comment by chharvey** [comment]
# High-Level Summary

## Features Already Present on CanIUse
- [CSS 2.1 Generated content for pseudo-elements](https://caniuse.com/#feat=css-gencontent)
- [CSS 2.1 Counters](https://caniuse.com/#feat=css-counters)

## CSS Generated Content Module Level 3
- [Spec](https://www.w3.org/TR/css-content-3/)

### Features In Spec
1. “Generated content should be searchable, selectable, and available to assistive technologies”
2. `content` property alt text: `content: url('./img/star.png') / 'New!';`
3. `content` property keyword values: `normal | none | contents`
4. `leader()` function: `leader(dotted | solid | space | <string>)`
5. `target-counter()`, `target-counters()`, `target-text()` functions
6. `string-set` property
7. `string()` function
8. `content()` function (not to be confused with `content` property)

**Comment by caesar** [comment]
+1

**Comment by manuelmeister** [comment]
I still think this is relevant today

**Comment by vintprox** [comment]
Browser typesetting is so behind LaTeX/ConTeXt, we don't even get mention of it ([6 years old exception](https://www.smashingmagazine.com/2015/01/designing-for-print-with-css/)).

**Comment by EdgarSchouten** [comment]
+1

**Comment by NigelCunningham** [comment]
+1

**Comment by WebDragon** [comment]
+1 just came 'round looking for this for generating an Annual Directory from html to pdf and was somewhat astonished to learn it hadn't yet been implemented

**Comment by MartinMerkel** [comment]
0Being able to use target-counter(s and target-text for online content and not just printed media would be extremely helpful.

**Comment by RobMayer** [comment]
+1

**Comment by SimonEast** [comment]
So `target-counter()` is in the CSS3 content draft spec (https://www.w3.org/TR/css-content-3/#cross-references) but not visible anywhere on caniuse.com. Even if zero browsers actively support it, can that still be shown on caniuse so developers can be informed?

Perhaps with a footnote that some PDF and ebook generators do support it.
